impl/src/main/java/io/opencensus/impl/internal/DisruptorEventQueue.java

```diff
@@ -134,13 +154,31 @@ public static DisruptorEventQueue getInstance() {
    */
   @Override
   public void enqueue(Entry entry) {
-    long sequence = ringBuffer.next();
-    try {
-      DisruptorEvent event = ringBuffer.get(sequence);
-      event.setEntry(entry);
-    } finally {
-      ringBuffer.publish(sequence);
-    }
+    enqueuer.enqueue(entry);
+  }
+
+  /** Shuts down the underlying disruptor. */
+  @Override
+  public void shutdown() {
+    enqueuer =
+        new DisruptorEnqueuer() {
+          boolean logged = false;
+
+          @Override
+          public void enqueue(Entry entry) {
+            if (!logged) {
+              logged = true;
```

Comment:
This access to `logged` isn't synchronized.  I'm not sure whether it would be better to use a lock, a volatile variable, or AtomicBoolean.  Only using volatile would allow duplicate log statements, but that probably isn't an issue.